src/autoloads/transition_handler.gd

```diff
@@ -2,61 +2,50 @@ extends Node
 
 enum States {
 	MENU			# Not in game
+	WAITING			# Between states
 	LOBBY			# In the lobby where people can join
 	MAP				# On a game map
 }
 
-var gameScene: Node
-var mainMenuScene: Node
-var appearanceScene: Node
-var currentScene: Node
 var currentState: int = States.MENU setget toss, getCurrentState
-
-func _ready() -> void:
-	## Preload the game scene
-	var root: Node = get_tree().get_root()
-	var game: Resource = ResourceLoader.load("res://game/game.tscn")
-	## Save current (main menu) scene
-	mainMenuScene = root.get_child(root.get_child_count() - 1)
-	## Instantiate and save th game scene
-	gameScene = game.instance()
-	## Current scene is the main menu
-	currentScene = mainMenuScene
-
-func switchScene(nextScene: Node) -> void:
-	## Removes current scene from scene tree
-	var root: Node = get_tree().get_root()
-	root.remove_child(currentScene)
-	## Sets the next scene as the current scene
-	root.add_child(nextScene)
-	currentScene = nextScene
-	get_tree().set_current_scene(currentScene)
+var gameScene: Node2D setget toss, getGameScene
 
 func toss(_newValue) -> void:
 	pass
 
 func showMainMenu() -> void:
 	## Switch to main menu scene
-	switchScene(mainMenuScene)
+	Scenes.switchBase("res://ui_elements//main_menu.tscn", true)
```

Comment:
We probably want to switch base to menu_base, and overlay main menu.

Reply:
true, true.
luckily, nothing's calling this function yet.
will do.